tests/components/zwave_js/test_sensor.py

```diff
@@ -730,3 +731,54 @@ async def test_statistics_sensors(
             state = hass.states.get(f"{prefix}{suffix_key}")
             assert state
             assert state.state == str(val)
+
+
+ENERGY_PRODUCTION_ENTITY_MAP = {
+    "energy_production_power": {
+        "state": 1.23,
+        "attributes": {
+            "unit_of_measurement": UnitOfPower.WATT,
+            "device_class": SensorDeviceClass.POWER,
+            "state_class": SensorStateClass.MEASUREMENT,
+        },
+    },
+    "energy_production_total": {
+        "state": 1234.56,
+        "attributes": {
+            "unit_of_measurement": UnitOfEnergy.WATT_HOUR,
+            "device_class": SensorDeviceClass.ENERGY,
+            "state_class": SensorStateClass.TOTAL_INCREASING,
+        },
+    },
+    "energy_production_today": {
+        "state": 123.45,
+        "attributes": {
+            "unit_of_measurement": UnitOfEnergy.WATT_HOUR,
+            "device_class": SensorDeviceClass.ENERGY,
+            "state_class": SensorStateClass.TOTAL,
+        },
+    },
+    "energy_production_time": {
+        "state": 123456.0,
+        "attributes": {
+            "unit_of_measurement": UnitOfTime.SECONDS,
+            "device_class": SensorDeviceClass.DURATION,
+            "state_class": None,
+        },
+    },
+}
+
+
+async def test_energy_production_sensors(
+    hass: HomeAssistant, energy_production, client, integration
+) -> None:
+    """Test sensors for Energy Production CC."""
+    for entity_id_suffix, state_data in ENERGY_PRODUCTION_ENTITY_MAP.items():
+        state = hass.states.get(f"sensor.node_2_{entity_id_suffix}")
+        assert state
+        assert state.state == str(state_data["state"])
+        for attr, val in state_data["attributes"].items():
+            if val is None:
+                assert attr not in state.attributes
```

Comment:
Please make a separate parameter container for values that should not be in the state attributes if we want to assert that too. We don't want conditions in the tests.